packages/kbn-unified-data-table/src/components/data_table.test.tsx

```diff
@@ -846,4 +846,32 @@ describe('UnifiedDataTable', () => {
       expect(getFieldColumns()).toEqual(['@timestamp', '_index', 'bytes', 'extension', 'message']);
     });
   });
+  describe('columnHeaders custom column labels', () => {
+    const mockColumnHeaders: Record<string, string> = {
+      test_column_1: 'test_column_one',
+      test_column_2: 'test_column_two',
+    } as const;
+
+    it('Columns that has corresponding custom label in columnHeaders props should show the label name for the column name', async () => {
+      renderDataTable({
+        columns: ['test_column_1', 'test_column_2'],
+        columnHeaders: mockColumnHeaders,
+      });
+      expect(getColumnHeaders().includes('test_column_one')).toBeTruthy();
+      expect(getColumnHeaders().includes('test_column_two')).toBeTruthy();
+      expect(getColumnHeaders().includes('test_column_1')).toBeFalsy();
+      expect(getColumnHeaders().includes('test_column_2')).toBeFalsy();
+    });
+
+    it('Columns that do not have corresponding custom label in columnHeaders props should show the original column name', async () => {
+      renderDataTable({
+        columns: ['test_column_1', 'test_column_3'],
+        columnHeaders: mockColumnHeaders,
+      });
+      expect(getColumnHeaders().includes('test_column_one')).toBeTruthy();
+      expect(getColumnHeaders().includes('test_column_two')).toBeFalsy();
+      expect(getColumnHeaders().includes('test_column_1')).toBeFalsy();
+      expect(getColumnHeaders().includes('test_column_3')).toBeTruthy();
+    });
+  });
```

Comment:
those tests look good, but since the main logic is on the `data_table_columns.tsx` file I would suggest we move them to the `data_table_columns.test.tsx` file